### PR TITLE
Restore original set of letters from assets when changing language

### PR DESCRIPTION
**Pull request by zenseii**
This PR aims to solve an issue described in https://github.com/ihhub/fheroes2/issues/5187#issuecomment-1086843531 and which is also the root problem for why the changes made by `generateFrenchAlphabet()` to the standard set of letters presist across all languages when changing to any other language other than either English or the language present in the assets, like when having German original assets present, after French has been used as language.

The issue can be observed in any current fheroes2 release by looking at the English Clouds of Xeen map description and switching between French and other languages. In French the key `&` is replaced by `ù`. You would need other assets than the French to test this.

Original description with `&` intact, because French has not been set:
image

Language set to any other language than either English or the one provided with assets (in my case German assets are present) after having set language to French at any point. Polish is used here:
image

The issue also presents itself when trying to type the keys `" # ¤ &` etc. and they get replaced with `à ô û ù` etc. after French language has been set at some point.
Polish language being used after French and letters `" # ¤` have been typed:
image

This issue of typing letters is also fixed by the changes in this PR.

Here the lowercase normal font k does not change when Norwegian language is set, and you can see that `Norsk` aka. Norwegian uses its lowercase k by comparing it to the one in `polski`. Russian small k is also preserved:
image

Compared to before:
image

### CODE REVIEW

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -1626,6 +1626,9 @@ namespace
 
     void generateAlphabet( const fheroes2::SupportedLanguage language )
     {
+        // Restore original 0-96 letters when changing language in case of changes to them.
+        alphabetPreserver.restore();
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:1630]: We already have a function which calls this function:
```cpp
void updateAlphabet( const SupportedLanguage language, const bool loadOriginalAlphabet )
        {
            if ( loadOriginalAlphabet || !isAlphabetSupported( language ) ) {
                alphabetPreserver.restore();
            }
            else {
                alphabetPreserver.preserve();
                generateAlphabet( language );
            }
        }
```
Wouldn't it better to put the call just after `alphabetPreserver.preserve();`?

File: src/fheroes2/agg/agg_image.cpp [hunk 2]
```diff
@@ -1626,6 +1626,9 @@ namespace
 
     void generateAlphabet( const fheroes2::SupportedLanguage language )
     {
+        // Restore original 0-96 letters when changing language in case of changes to them.
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:1629]: This comment is not entirely correct: restoring the original alphabet might restore more than 96 letters. Please try German for example.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I left few comments here. Also this approach is not ideal: what if we use French assets? Will the problem be there?

File: src/fheroes2/agg/agg_image.cpp (hunk 1 above)
Comment (zenseii) [reply src/fheroes2/agg/agg_image.cpp:1846]: Yes, this is a much better placement. Not sure why I didn't consider this first. I've tested it and it works.

**Comment by zenseii** [comment]
Hi @ihhub ! Thanks for the review. I've adressed the comments and tested the PR with German and French assets and the problem isn't observed anymore.

**Comment by ihhub** [comment]
@zenseii , a bunch of thanks for fixing this issue!

**Pull request merged**